src/cockpit/config.py

```diff
@@ -24,11 +24,25 @@
 
 logger = logging.getLogger(__name__)
 
-ETC_COCKPIT = Path('/etc/cockpit')
 XDG_CONFIG_HOME = Path(os.getenv('XDG_CONFIG_HOME') or os.path.expanduser('~/.config'))
 DOT_CONFIG_COCKPIT = XDG_CONFIG_HOME / 'cockpit'
 
 
+def lookup_config(filename):
```

Comment:
Can you restructure this to take no arguments, return a Path object, and decorate it with `@functools.lru_cache`?  You need to make sure to call `.cache_clear()` from test cases, though...

I don't like the idea that what we consider the cockpit path could change during program execution, and it seems a bit overkill to do this calculation each time.

Reply:
How would that work with no arguments? This searches all paths in X_C_H for the requested file/path, it's not a static thing.

Reply:
I got confused by your "take no arguments". I rewrote this using lru_cache (TIL!), returning `Path`, and adding type annotations. It's nicer  now!